### ISSUE TITLE
Missing atomic operations

### ISSUE THREAD

**Issue by bordoley** [ocamllabs/ocaml-multicore#196]
This is mostly a hypothetical request, but in general it would be useful to provide additional atomic operations including:
```
(* return the previous value when setting a new value *)
val exchange : 'a t -> 'a -> 'a

(* increment an int atomically *)
val incr: int t

(* decrement an int atomically *)
val decr: int t
```

**Comment by stedolan** [comment]
I agree, these are very useful.

**Comment by bordoley** [comment]
Somewhat related, are you considering any type of API for performing atomic operations upon mutable record fields? C# enables this via the Interlocked class and allows passing a direct pointer to a class field via the `ref` keyword. This might a useful feature for performance sensitive code that seeks to avoid allocations by packing multiple atomic values into the same record.

**Comment by stedolan** [comment]
Not at the moment, no. There's no analogue of C#'s `ref` in OCaml, and we're not currently planning to add one.

**Comment by stedolan** [comment]
Closed by #225
